server/mysterium_api.go

```diff
@@ -17,105 +15,111 @@ const (
 	mysteriumApiLogPrefix = "[Mysterium.api] "
 )
 
-var mysteriumApiUrl string
+//HttpTransport interface with single method do is extracted from net/transport.Client structure
+type HttpTransport interface {
+	Do(*http.Request) (*http.Response, error)
+}
 
 type mysteriumApi struct {
-	json JsonClient
+	http HttpTransport
 }
 
 func NewClient() Client {
 	return &mysteriumApi{
-		json: NewJsonClient(
-			mysteriumApiUrl,
-			&http.Client{
-				Transport: &http.Transport{},
-			}),
+		&http.Client{
+			Transport: &http.Transport{},
+		},
 	}
 }
 
-func (mApi *mysteriumApi) RegisterIdentity(identity identity.Identity) (err error) {
-	response, err := mApi.json.Post("identities", dto.CreateIdentityRequest{
+func (mApi *mysteriumApi) RegisterIdentity(identity identity.Identity) error {
+	req, err := newPostRequest("identities", dto.CreateIdentityRequest{
 		Identity: identity.Address,
 	})
+	if err != nil {
+		return err
+	}
 
+	resp, err := mApi.http.Do(req)
 	if err == nil {
-		defer response.Body.Close()
+		defer resp.Body.Close()
 		log.Info(mysteriumApiLogPrefix, "Identity registered: ", identity)
 	}
-
-	return
+	return err
 }
 
-func (mApi *mysteriumApi) NodeRegister(proposal dto_discovery.ServiceProposal) (err error) {
-	response, err := mApi.json.Post("node_register", dto.NodeRegisterRequest{
+func (mApi *mysteriumApi) NodeRegister(proposal dto_discovery.ServiceProposal) error {
+	req, err := newPostRequest("node_register", dto.NodeRegisterRequest{
 		ServiceProposal: proposal,
 	})
+	if err != nil {
+		return err
+	}
 
+	resp, err := mApi.http.Do(req)
```

Comment:
This block could be DRYied:
```
resp, err := mApi.http.Do(req)
if err == nil {
    defer resp.Body.Close()
}
```
Similar places was refactored in `ipify.doRequest(reqest, responseDto)`

Reply:
I will check it